scripts/yul_coverage.sh

```diff
@@ -0,0 +1,145 @@
+#
+# The documentation for solidity is hosted at:
+#
+#     https://solidity.readthedocs.org
+#
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2020 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -e
+
+ROOT_DIR="$(dirname "$0")"/..
+
+for arg in "$@"; do
+  case "$arg" in
+  --no-stats) NO_STATS=1 ;;
+  --successful) SHOW_SUCCESSFUL=1 ;;
+  --internal-compiler-errors) SHOW_INTERNAL_COMPILER_ERRORS=1 ;;
+  --unimplemented-feature-errors) SHOW_UNIMPLEMENTED_FEATURE_ERRORS=1 ;;
+  --other-errors) SHOW_OTHER_ERRORS=1 ;;
+  --list-files) ONLY_LIST_FILES=1 ;;
+  *)
+    echo "Usage:"
+    echo "  $(basename "${0}") [--no-stats] [--successful] [--internal-compiler-errors] [--unimplemented-feature-errors] [--other-errors] [--list-files]"
+    echo "  --no-stats                     will not print the stats to stdout"
+    echo "  --successful                   print output of successful test-case compilations to stdout"
+    echo "  --internal-compiler-errors     print output of test-case compilations that resulted in"
+    echo "                                 internal compilation errors to stdout"
+    echo "  --unimplemented-feature-errors print output of test-case compilations that resulted in"
+    echo "                                 unimplemented feature errors to stdout"
+    echo "  --other-errors                 print output of test-case compilations that resulted in"
+    echo "                                 errors that where not internal compiler errors or unimplemented feature errors"
+    echo "                                 to stdout"
+    echo "  --list-files                   will not print the compiler output to stdout, it will just print the files"
+    echo "                                 e.g. './yul_coverage.sh --successful --list-files' will just return a list of"
+    echo "                                 files where it's compilation result was successful"
+    exit 0
+    ;;
+  esac
+done
+
+show_output_if() {
+  local VAR=${1}
+  if [ -n "${VAR}" ]; then
+    echo "${SOL_FILE}"
+    if [ -z "${ONLY_LIST_FILES}" ]; then
+      echo "${OUTPUT}"
+      echo ""
+    fi
+  fi
+}
+
+FAILED=()
+SUCCESS=()
+SOLC=${SOLC:-"$(which solc)"}
```

Comment:
I wonder whether this is a good way to find ``solc``... it should at least be documented at the top that it takes ``solc`` from the PATH and not necessarily from the repo this is run in... alternatively you could make it an argument or default to `${ROOT_DIR}/build/solc/solc` (while providing a way to change the path with an argument).
I'd argue that the ``solc`` binary you want to test is actually usually *not* the one found in PATH... I'd be fine with not changing it, but only documenting this, as well, though.

Reply:
Well, the new version basically does the same... but maybe it's ok, since the environment variable is documented, even though it doesn't specify what's used by default...